### DISCUSSION TITLE
Custom package name for Hex dependency

### DISCUSSION THREAD

**Discussion by bernardo-martinez** [renovatebot/renovate#32834] [Suggest an Idea]
### Tell us more.

**What would you like Renovate to be able to do?**

In Mix in elixir you can setup a dependency with a different name than the default one fetched from hex.pm, see: https://hex.pm/docs/usage#options

For example:
```
{:gun, "~> 2.0.0", hex: :grpc_gun},
```

This will fetch the `grpc_gun` dependency from hex.pm located here: https://hex.pm/packages/grpc_gun but will treat it locally as `gun`

**If you have any ideas on how this should be implemented, please tell us here.**

I think the mix manager should have an additional regex to search for such `hex:` option and use it to set `PackageDependency.sourceUrl` accordingly, which then would be used in the hex versioning?

**Is this a feature you are interested in implementing yourself?**
Yes, just need some discussion and guidance first

**Comment by bernardo-martinez** [comment]
I have created a minimal reproduction repo here: https://github.com/bernardo-martinez/renovate-reproduction-32834/pull/1

**Reply by rarkins** [reply]
Please follow the template readme instructions. It's also not a reproduction in the first place if you create the PR by hand instead of install the bot

**Reply by bernardo-martinez** [reply]
ok thanks i will read again, the readme instructions. OTOH I did not created it by hand, i mean i locally ran in `main`: `pnpm start bernardo-martinez/renovate-reproduction-32834` with my github access token, which is what generated the PR 🤔  isn't this a valid way to run it?

**Reply by RahulGautamSingh** [reply]
It is but sometimes you file config (`config.js`) might be affecting the outcome. So to be safe we insist to install the hosted app

**Reply by rarkins** [reply]
It's missing all the usual PR body of a renovate PR

**Reply by bernardo-martinez** [reply]
thanks for the answers guys. I have updated the README as required, and also created a new PR: https://github.com/bernardo-martinez/renovate-reproduction-32834/pull/3 (it can be seen that the author of the commit is the renovate-bot) I created it again from my local (dunno how to do `"we insist to install the hosted app" 🙇 `) and it has now the usual PR body, lemme know if now is ok 🙏

**Reply by rarkins** [reply]
Thanks, converted to issue here: https://github.com/renovatebot/renovate/issues/32972
